LookingMenu/Scenes/Detail Screen/DetailRecipeController.swift

```diff
@@ -0,0 +1,165 @@
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        imageRecipe.layer.cornerRadius = imageRecipe.frame.width / 2
+    }
+    
+    private func addEventLabel() {
+        let tapEquip = UITapGestureRecognizer(target: self,
+                                              action: #selector(self.clickLabel(_:)))
+        let tapIngre = UITapGestureRecognizer(target: self,
+                                              action: #selector(self.clickLabel(_:)))
+        labelEquip.isUserInteractionEnabled = true
+        labelEquip.addGestureRecognizer(tapEquip)
+        labelIngre.isUserInteractionEnabled = true
+        labelIngre.addGestureRecognizer(tapIngre)
+    }
+    
+    @objc private func clickLabel(_ sender: UITapGestureRecognizer) {
+        guard let textLabel = sender.view  else { return }
+        let checkLabel = textLabel == labelEquip
+        labelEquip.textColor = checkLabel ? .redDesign : .blackDesign
+        labelIngre.textColor = checkLabel ? .blackDesign : .redDesign
+        constraintBottomBarViewCenterEquip.priority = checkLabel ? .defaultHigh : .defaultLow
+        constraintBottomBarViewCenterIngre.priority = checkLabel ? .defaultLow : .defaultHigh
+        UIView.animate(withDuration: 0.5,
+                       delay: 0.0,
+                       options: .curveEaseOut, animations: {
+                        self.view.layoutIfNeeded()
+                        self.EquipmentTB.alpha = checkLabel ? 1.0 : 0.0
+                        self.IngredientTB.alpha = checkLabel ? 0.0 : 1.0
+                       }, completion: nil)
+    }
+    
+    @IBAction private func goViewTextRecipe(_ sender: Any) {
+        guard let textVC = storyboard?
+                .instantiateViewController(
+                    withIdentifier: IdStoryBoardViews.textRecipe)
+                as? TextRecipeController else { return }
+        textVC.recipe = recipe
+        navigationController?.pushViewController(textVC, animated: true)
+    }
+    
+    @IBAction private func goViewVideoRecipe(_ sender: Any) {
+        guard let videoVC = self.storyboard?
+                .instantiateViewController(
+                    withIdentifier: IdStoryBoardViews.videoRecipe)
+                as?  VideoRecipeController else { return }
+        videoVC.recipeFromDetail = recipe
+        navigationController?.pushViewController(videoVC, animated: true)
+    }
+}
+
+extension DetailRecipeController : UITableViewDelegate,
+                                   UITableViewDataSource {
+    func tableView(_ tableView: UITableView,
+                   numberOfRowsInSection section: Int) -> Int {
+        if tableView == self.IngredientTB {
+            return ingredients.count
+        } else {
+            return equipments.count
+        }
+    }
+    
+    func tableView(_ tableView: UITableView,
+                   heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return ConstantDetailRecipeView.heightTableCell
```

Comment:
dùng `UITableView.autoDimension`. Estimate height thì mới nên set cứng. Như này lên những màn nhỏ sẽ rất to